plan/relations.go

```diff
@@ -585,9 +585,9 @@ const (
 	JoinTypeOuter       = proto.JoinRel_JOIN_TYPE_OUTER
 	JoinTypeLeft        = proto.JoinRel_JOIN_TYPE_LEFT
 	JoinTypeRight       = proto.JoinRel_JOIN_TYPE_RIGHT
-	JoinTypeSemi        = proto.JoinRel_JOIN_TYPE_SEMI
-	JoinTypeAnti        = proto.JoinRel_JOIN_TYPE_ANTI
-	JoinTypeSingle      = proto.JoinRel_JOIN_TYPE_SINGLE
+	JoinTypeRightSemi   = proto.JoinRel_JOIN_TYPE_RIGHT_SEMI
```

Comment:
it looks like some other changes are in this patch. Maybe it needs to be rebased?

Reply:
This is needed because of proto update to v.54. [This](https://github.com/substrait-io/substrait/commit/bed84ecb6193c22bf2ff83dc3a391ec5a9a3aa68) PR updated these types

Reply:
Best to do that as a separate patch.

Reply:
Raised [this](https://github.com/substrait-io/substrait-go/pull/48) PR for proto update. Once it is merged, I will rebase this PR.